packages/components/controlled-input/src/ControlledInput.tsx

```diff
@@ -0,0 +1,163 @@
+import React, {
+  HTMLProps,
+  EventHandler,
+  ChangeEvent,
+  FocusEvent,
+  KeyboardEvent,
+  useCallback,
+  useRef,
+  useEffect,
+} from 'react';
+import { cx } from 'emotion';
+import { Done, Minus } from '@contentful/f36-icons';
+import type { IconProps } from '@contentful/f36-icons';
+
+import { styles } from './ControlledInput.styles';
+import { Box } from '@contentful/f36-core';
+
+export interface ControlledInputProps extends HTMLProps<HTMLInputElement> {
+  id?: string;
+  required?: boolean;
+  labelText: string;
+  checked?: boolean;
+  onChange?: EventHandler<ChangeEvent<HTMLInputElement>>;
+  name?: string;
+  onBlur?: EventHandler<FocusEvent<HTMLInputElement>>;
+  onFocus?: EventHandler<FocusEvent<HTMLInputElement>>;
+  value?: string;
+  disabled?: boolean;
+  type?: 'checkbox' | 'radio';
+  className?: string;
+  testId?: string;
+  willBlurOnEsc?: boolean;
+  indeterminate?: boolean;
+  ref?: React.Ref<HTMLInputElement>;
+}
+
+// Use forwarded ref with React hooks
+// Src: https://itnext.io/reusing-the-ref-from-forwardref-with-react-hooks-4ce9df693dd
+const useCombinedRefs = <T extends HTMLElement>(
```

Comment:
Handling a `ref` which is used for `forwardRef` but also in Reak hooks requires this merging logic. Any feedback or alternative solution is welcome :)

Reply:
Yeah this is the way to go, but actually, in this particular case it's not necessary since we always want to forward the reference to our root element, in this case the `Box`. We've yet to decide on a good way we can expose inner element references.